### PR TITLE
Removes peer dependency on react-test-renderer.

### LINKED ISSUE #20: Remove peer dependency on `react-test-renderer`
I don't use React in my project, but I get a noisy peer dep warning from both npm and yarn when installing this module.

As it's optional, do you think you could drop it as a peer dep?

### PR DESCRIPTION
**Pull request by kpsroka**
Fixes #20

### CODE REVIEW

File: __tests__/snapshotDiff.test.js [hunk 1]
```diff
@@ -149,3 +149,26 @@ test('can use contextLines with React components', () => {
...
+
+  afterEach(() => {
+    jest.resetModules();
+  });
+
+  test('throws with sensible message on missing react-test-renderer', () => {
+    const testComponentA = <Component test="a" />;
+    const testComponentB = <Component test="b" />;
+    try {
+      snapshotDiff(testComponentA, testComponentB);
```
Comment (SimenB) [inline __tests__/snapshotDiff.test.js]: `expect(snapshotDiff(testComponentA, testComponentB)).toThrowErrorMatchingSnapshot()`

File: src/index.js [hunk 2]
```diff
@@ -58,8 +58,20 @@ function diffStrings(valueA: any, valueB: any, options: Options) {
   });
 }
 
+function requireReactTestRenderer() {
+  try {
+    return require('react-test-renderer'); // eslint-disable-line import/no-extraneous-dependencies
+  } catch (error) {
```
Comment (SimenB) [inline src/index.js]: maybe check for `e.code === 'MODULE_NOT_FOUND'?

File: src/index.js [hunk 3]
```diff
@@ -58,8 +58,20 @@ function diffStrings(valueA: any, valueB: any, options: Options) {
...
+    throw new Error(
+      `Failed to load optional module "react-test-renderer". ` +
+        `Are you sure that it's listed in your project's dependencies?\n` +
```
Comment (SimenB) [inline src/index.js]: Maybe say something like "if you need to compare React elements..."?

**Comment by kpsroka** [comment]
Thanks! PTAL.

**Pull request merged**